File: ems-backend/src/main/java/com/rizwan/ems_backend/service/DepartmentService.java
package com.rizwan.ems_backend.service;

import java.util.List;

import com.rizwan.ems_backend.entity.Department;

public interface DepartmentService {

    Department saveDepartment(Department department);

    List<Department> getAllDepartments();

    Department getDepartmentById(Long id);

    Department updateDepartment(Long id, Department department);

    void deleteDepartment(Long id);
}

File: ems-frontend/src/services/DepartmentService.js
import axios from "axios";

const REST_API_BASE_URL = "http://localhost:8080/api/departments";

class DepartmentService {

    getAllDepartments() {
        return axios.get(REST_API_BASE_URL);
    }

    getDepartmentById(id) {
        return axios.get(REST_API_BASE_URL + "/" + id);
    }

    createDepartment(department) {
        return axios.post(REST_API_BASE_URL, department);
    }

    updateDepartment(id, department) {
        return axios.put(REST_API_BASE_URL + "/" + id, department);
    }

    deleteDepartment(id) {
        return axios.delete(REST_API_BASE_URL + "/" + id);
    }

}

export default new DepartmentService();